downloads/submodule-vs-subtree-cheat-sheet.md

```diff
@@ -0,0 +1,59 @@
+# Handling Dependencies with Submodules and Subtrees
+
+Submodules and subtrees are git tools that allow subprojects to be included as a subdirectory within a project. The implementation of each is very different.
+
+### Adding a New Subproject
+
+| Submodule | Subtree |
+| ---------- | ---------- |
+| `git submodule add https://github.com/githubtraining/example-submodule` <br> `git commit -m “adding new submodule”`| `git subtree add --prefix=example-submodule https://github.com/githubtraining/example-submodule master --squash` |
+| The submodule command adds a new file called `.gitmodules` along with a sub-directory containing the files from `example-submodule`. Both are added to your index and you simply need to commit them. The subproject's history remains independent of the container project. | The subtree command adds a sub-directory containing the files from `example-submodule`. The `--squash` option combines the sub-project history into a single commit and grafts it onto your existing tree. You can omit the squash option to maintain all of the history from the designated branch of the subtree. |
+
+### Viewing a Diff of the Subproject
+
+| Submodule | Subtree |
+| ---------- | ---------- |
+| To view a diff of the submodule: <br> `git diff --cached example-submodule` <br> `git diff --cached --submodule example-submodule` | No special command required |
```

Comment:
@crichID to view a diff of the submodule, wouldn't we just `cd` into that directory and use diff as usual? Maybe we can clarify and say that this command helps us see diffs for the submodule *after staging*?